### PR TITLE
WP index to the values widget

### PR DESCRIPTION
**Pull request by dakejahl**
Adds current WP index as a Vehicle Fact for use to display in the values widget.

### CODE REVIEW

**Comment by Antiheavy** [comment]
Addresses https://github.com/mavlink/qgroundcontrol/issues/7791

**Comment by DonLakeFlyer** [comment]
The naming for all of this should be "Mission Item Index" and variations of that. Indices are associated with all mission items not just waypoints.

File: src/Vehicle/Vehicle.cc [hunk 1]
```diff
@@ -3984,6 +3989,13 @@ void Vehicle::_updateHeadingToNextWP()
     }
 }
 
+void Vehicle::_updateWpIndexNumber()
+{
+    const int _currentIndex = _missionManager->currentIndex();
```
Comment (DonLakeFlyer) [inline src/Vehicle/Vehicle.cc]: This shouldn't use "_" for the variable name.

File: src/Vehicle/Vehicle.cc [hunk 2]
```diff
@@ -3984,6 +3989,13 @@ void Vehicle::_updateHeadingToNextWP()
...
+
+    _wpIndexNumberFact.setRawValue(_currentIndex + 1);
```
Comment (DonLakeFlyer) [inline src/Vehicle/Vehicle.cc]: The "+1" must be based on !firmwarePlugin()->sendHomePositionToVehicle(). If this is false then you +1 the index. This accounts for a different in how PX4 and ArduPilot handle mission items. ArduPilot always has a fake home position item in index 0, PX4 does not. QGC always has a home position item in index 0. So item indices coming directly from mavlink must be massaged appropriately.

**Comment by dakejahl** [comment]
Thanks for the review, I've updated the PR with the requested changes.

File: src/Vehicle/VehicleFact.json [hunk 3]
```diff
@@ -104,6 +104,12 @@
     "decimalPlaces":    1,
     "units":            "m"
 },
+{
+    "name":             "missionItemIndex",
+    "shortDescription": "Mission Item Index",
+    "type":             "uint16",
+    "units":            ""
```
Comment (DonLakeFlyer) [inline src/Vehicle/VehicleFact.json]: This should just be left out

**Comment by DonLakeFlyer** [comment]
Thanks Jake

File: src/Vehicle/VehicleFact.json [hunk 4]
```diff
@@ -104,6 +104,11 @@
     "decimalPlaces":    1,
     "units":            "m"
 },
+{
+    "name":             "missionItemIndex",
+    "shortDescription": "Mission Item Index",
+    "type":             "uint16",
```
Comment (DonLakeFlyer) [inline src/Vehicle/VehicleFact.json]: Unit tests are failing because there is an extra "," here

**Pull request merged**